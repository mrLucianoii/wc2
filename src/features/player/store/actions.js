import types from './constants';

function markInAction(milliseconds){
  return {
    type: types.MARK_IN,
    timeStamp: Math.round(milliseconds)
  }
}

function markOutAction(milliseconds){
  return {
    type: types.MARK_OUT,
    timeStamp: Math.round(milliseconds)
  }
}

function startPlaying(videoID) { // event passed onReady at component render
    return {
        type: types.PLAY_VIDEO,
        vidID: videoID
      }
  }
  
export default {
    markInAction,
    markOutAction,
    startPlaying,
}