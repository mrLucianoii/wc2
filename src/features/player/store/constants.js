const PLAY_VIDEO = 'weCliques-app/movies/PLAY_VIDEO';
const MOVIE_PARAM = 'weCliques-app/movies/MOVIE_PARAM';
const FETCHED_MOVIE_PROPS = 'weCliques-app/movies/FETCHED_MOVIE_PROPS';
const MARK_IN = 'weCliques-app/movies/MARK_IN';
const MARK_OUT = 'weCliques-app/movies/MARK_OUT';
const CREATED_SEGMENT = 'weCliques-app/movies/CREATED_SEGMENT';
const SEGMENT_SELECTED = 'weCliques-app/movies/SEGMENT_SELECTED';
const SEGMENT_ACTIVE = 'weCliques-app/movies/SEGMENT_ACTIVE';
const CHANGE_SEGMENT = 'weCliques-app/movies/CHANGE_SEGMENT';
const FULL_CLIP = 'weCliques-app/movies/FULL_CLIP';


export default {
    PLAY_VIDEO,
    MOVIE_PARAM,
    FETCHED_MOVIE_PROPS,
    MARK_IN,
    MARK_OUT,
    CREATED_SEGMENT,
    SEGMENT_SELECTED,
    SEGMENT_ACTIVE,
    CHANGE_SEGMENT,
    FULL_CLIP,
}