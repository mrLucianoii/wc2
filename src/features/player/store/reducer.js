import types from './constants';

const initialState = {
    selectedMovie: 'ELeMaP8EPAA',
    segmentStart: 0,
    segmentEnd: null,
};

export default (state = initialState, actions = {}) => {
  switch (actions.type) {
    case types.MARK_IN: {
      return {
        ...state,
        segmentStart: actions.timeStamp,
      }
    }
    default:

      return state;
  }
};
