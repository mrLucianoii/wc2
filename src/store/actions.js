const isLoading = () => (dispatch) => {
  dispatch({
    type: 'MOVIE_FINISHED',
  });
}; 

export default {
 isLoading,
};
