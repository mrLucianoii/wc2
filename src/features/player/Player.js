import React, { Component, PropTypes } from "react";
import { connect } from "react-redux";
import actions from "./store/actions";
import { Button } from "react-bootstrap";
import YouTube from "react-youtube";

export class Player extends Component {
  state = {
    player: null,
    duration: 0,
    segmentList: []
  };
  onReady = this.onReady.bind(this);
  onPlayVideo = this.onPlayVideo.bind(this);
  onPauseVideo = this.onPauseVideo.bind(this);

  onReady(event) {
    this.setState({
      player: event.target
    });
  }
  onPlayVideo() {
    const { startPlaying } = this.props.actions;
    const { selectedMovie, segmentStart } = this.props.movies;

    this.state.player.playVideo();
    startPlaying(selectedMovie);
  }
  seekSegment() {
    const { segmentStart } = this.props.movies;
    this.state.player.seekTo(segmentStart);
  }
  onPauseVideo() {
    this.state.player.pauseVideo();
  }

  render() {
    const { selectedMovie, segmentStart, segmentEnd } = this.props.movies;
    const { markInAction, markOutAction, checkMarks } = this.props.actions;
    const opts = {
      playerVars: {
        autoplay: 1,
        loop: 1,
        start: Number(segmentStart, 10),
        end: Number(segmentEnd, 10),
        modestbranding: 1,
        controls: 1,
        rel: 0
      }
    };
    return (
      <div style={{ width: 660, height: "auto" }}>
        <YouTube
          videoId={selectedMovie}
          onReady={this.onReady}
          opts={opts}
          sandbox="allow-same-origin"
        />

        {/* <Button onClick={this.play.bind(this)}>Play</Button>
            <Button>Pause</Button> */}
        <Button onClick={this.onPlayVideo}>Play</Button>
        <Button onClick={this.onPauseVideo}>Pause</Button>
        <Button onClick={this.seekSegment.bind(this)}>Segment Start</Button>

        <Button
          onClick={() => {
            markInAction(this.state.player.getCurrentTime());
            checkMarks();
          }}
        >
          In-Mark
        </Button>
        <Button
          onClick={() => {
            markOutAction(this.state.player.getCurrentTime());
            checkMarks();
          }}
        >
          Out-Mark
        </Button>
        <Button onClick={() => {}}>full-play</Button>
      </div>
    );
  }
}

export default connect(
  state => ({
    movies: state.playerReducer
  }),
  dispatch => ({
    actions: {
      checkMarks: () => dispatch(actions.checkMarks()),
      markInAction: milliseconds =>
        dispatch(actions.markInAction(milliseconds)),
      markOutAction: milliseconds =>
        dispatch(actions.markOutAction(milliseconds)),
      startPlaying: () => dispatch(actions.startPlaying())
    }
  })
)(Player);
