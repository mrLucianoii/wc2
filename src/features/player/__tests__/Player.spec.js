import React from "react";
import { Provider } from "react-redux";
import store from "../../../store";
import renderer from "react-test-renderer";
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Player from "../Player";
import YouTube from "react-youtube";

Enzyme.configure({ adapter: new Adapter() });
function setup() {
  const props = {
    selectedMovie: 'ELeMaP8EPAA'
  }

  const enzymeWrapper = mount( <Provider store={store()}>
  <Player {...props}/>
</Provider>)

  return {
    props,
    enzymeWrapper
  }
}

describe("Embeded Youtube Player", () => {
  it("Matches Snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store()}>
            <Player />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Receives default props", () => {
    const { enzymeWrapper } = setup()
    const selectedMovie = enzymeWrapper.find(Player).props()
    expect(selectedMovie).toStrictEqual({selectedMovie: 'ELeMaP8EPAA'})
  })

  it("YouTube component receives default movie id", () => {
    const { enzymeWrapper } = setup()
    const PlayerWrapper = enzymeWrapper.find(Player)
    const YouTubeWrapper = PlayerWrapper.find(YouTube).props()

    expect(YouTubeWrapper.videoId).toStrictEqual('ELeMaP8EPAA');
  })
});
