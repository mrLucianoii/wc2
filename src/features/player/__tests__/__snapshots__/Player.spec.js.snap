// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embeded Youtube Player Matches Snapshot 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "width": 660,
    }
  }
>
  <div
    className=""
  >
    <div
      className={null}
      id={null}
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Pause
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Segment Start
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    In-Mark
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Out-Mark
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    full-play
  </button>
</div>
`;
